code/datums/diseases/_disease.dm

```diff
@@ -90,12 +91,12 @@ var/list/diseases = subtypesof(/datum/disease)
 	if(!(disease_flags & CURABLE))
 		return 0
 
-	. = 1
+	. = cures.len
 	for(var/C_id in cures)
-		if(!affected_mob.reagents.has_reagent(C_id))
+		if(!affected_mob.has_reagent(C_id))
 			.--
-			break //One missing cure is enough to fail
-
+	if(!. || (needs_all_cures && . < cures.len))
+	return 0
```

Comment:
Bad indentation. Also, use neersighted's way instead. WAY easier to read.
https://www.irccloud.com/pastebin/K2ncYgIr/